framework/PostProcessors/ValueDuration.py

```diff
@@ -0,0 +1,168 @@
+      In here the working dir is grepped.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def inputToInternal(self, currentInp):
+    """
+      Method to convert an input object into the internal format that is
+      understandable by this pp.
+      In this case, we only want data objects!
+      @ In, currentInp, object, an object that needs to be converted
+      @ Out, inputDict, dict, dictionary of the converted data
+    """
+    if len(currentInp) > 1:
+      self.raiseAnError(IOError, 'Expected 1 input HistorySet, but received {} inputs!'.format(len(currentInp)))
+    currentInp = currentInp[0]
+    if currentInp.type not in ['HistorySet']:
+      self.raiseAnError(IOError, 'ValueDuration postprocessor "{}" requires a HistorySet input! Got "{}".'
+                                 .format(self.name, currentInput.type))
+    return currentInp
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action.
+      @ In, inputIn, object, object contained the data to process. (inputToInternal output)
+      @ Out, realizations, list, list of realizations obtained
+    """
+    inData = self.inputToInternal(inputIn)
+    realizations = []
+    # new load duration curve for each sample
+    for s in range(len(inData)):
+      # obtain relevant sample
+      sample = inData.realization(index=s)
+      # solutions for each realization are stored in this dict
+      rlz = {}
+      # separate curve for each target
+      for target in self.targets:
+        data = sample[target]
+        # bin values
+        counts, edges = np.histogram(data,self.numBins)
+        ## reverse order of histogram, edges to get load duration axes
+        counts = counts[::-1]
+        edges = edges[::-1]
+        ## cumulatively stack counts, starting with highest value bin
+        cumulative = np.cumsum(counts)
+
+        # store results
+        rlz['counts_'+target] = cumulative
+        ## only keep upper value of edges so lengths match
+        rlz['bins_'+target] = edges[1:]
+
+        realizations.append(rlz)
+
+    return realizations
+
+  def collectOutput(self, finishedJob, output):
+    """
+      Function to place all of the computed data into the output object
+      @ In, finishedJob, JobHandler External or Internal instance, A JobHandler object that is in charge of running this post-processor
+      @ In, output, dataObjects, The object where we want to place our computed results
+      @ Out, None
+    """
+    evaluation = finishedJob.getEvaluation()
+    if isinstance(evaluation, Runners.Error):
+      self.raiseAnError(RuntimeError, "No available output to collect!")
+
+    realizations = evaluation[1]
+    for rlz in realizations:
+      output.addRealization(rlz)
```

Comment:
maybe a method can be created in the DataObject to handle multiple realizations.

Reply:
Possibly.  The urgency of this merge suggest it shouldn't be put off for such an extension, though.